csrc/python_frontend/fusion_definition.h

```diff
@@ -79,6 +64,23 @@ struct TORCH_CUDA_CU_API Scalar {
   //! A unique index to identifiy each recorded state item.
   size_t index;
 
+  //! Pointer to the FusionDefinition used to create this scalar
+  //! The FusionDefinition pointer is necessary to enable special
+  //! dunder operations (ie __add__()) from the python API.
+  FusionDefinition* fusion_definition;
+};
+
+struct TORCH_CUDA_CU_API Vector {
```

Comment:
It's probably in the follow up PR, so I won't nitpick on this.

But my instinct here is that we'll want to access individual elements within `Vector`, so maybe we'd prefer to have the length explicitly exposed (similar to how we have `dims` in Tensor).
I guess I'm really saying, we might want to have an `Array` instead of `Vector` here :stuck_out_tongue: 

But again, we don't have to go over this right here right now. Let's discuss that in the follow up PRs.

Reply:
Hahaha, Jacob is going fancy there :+1: 

I'm more thinking about just a static length `Array` and keeping nested container out of the scope as well. I feel those would significantly add complexity here and I need to see good examples to justify the work.